tests/phpunit/tests/query/wpLoop.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/**
+ * Tests to make sure looping through posts using the wp_loop() function works properly.
+ *
+ * @ticket 48193
+ *
+ * @group  query
+ */
+class Tests_WP_Loop extends WP_UnitTestCase {
+
+	/**
+	 * Create test posts in database.
+	 *
+	 * @param WP_UnitTest_Factory $factory
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		$factory->post->create( [ 'post_title' => 'Post 1' ] );
+		$factory->post->create( [ 'post_title' => 'Post 2' ] );
+		$factory->post->create( [ 'post_title' => 'Post 3' ] );
+	}
+
+	/**
+	 * Test iterating over the global WP_Query instance.
+	 */
+	public function test_global_query() {
+
+		// Ensure global query returns results in order by title.
+		add_filter( 'pre_get_posts', function ( WP_Query $query ) {
+			if ( $query->is_main_query() ) {
+				$query->set( 'orderby', 'title' );
+				$query->set( 'order', 'ASC' );
+			}
+		} );
+
+		// Required to setup global WP_Query instance
+		$this->go_to( '/' );
+
+		$i = 0;
+		foreach ( wp_loop() as $post ) {
+			$i ++;
+			$this->assertTrue( is_a( $post, 'WP_Post' ) );
+			$this->assertEquals( "Post {$i}", $post->post_title );
+			$this->assertEquals( "Post {$i}", get_the_title() );
+		}
+
+	}
+
+	/**
+	 * Test iterating over a custom WP_Query instance.
+	 */
+	public function test_custom_query() {
+
+		// Create query
+		$query = new WP_Query( [
+			'post_type' => 'post',
+			'order_by'  => 'title',
+			'order'     => 'ASC',
+		] );
```

Comment:
Use a data provider to pass in the different object types to a single test function.

Reply:
Unable to use a data provider in this case. PHPUnit docs: "All data providers are executed before both the call to the setUpBeforeClass static method and the first call to the setUp method. Because of that you can't access any variables you create there from within a data provider. This is required in order for PHPUnit to be able to compute the total number of tests."